src/main/java/org/opentripplanner/model/plan/ItinerariesCalculateLegTotals.java

```diff
@@ -45,19 +45,24 @@ private void calculate(List<Leg> legs) {
       } else if (leg.isStreetLeg()) {
         nonTransitDuration = nonTransitDuration.plus(dt);
         nonTransitDistanceMeters += leg.getDistanceMeters();
+      } else if (leg instanceof UnknownTransitPathLeg unknownTransitPathLeg) {
+        nTransitLegs += unknownTransitPathLeg.getNumberOfTransfers() + 1;
       }
+
       if (!leg.isWalkingLeg()) {
         walkOnly = false;
       }
+
       if (!leg.isStreetLeg()) {
         this.streetOnly = false;
       }
+
       if (leg.getElevationProfile() != null) {
         var p = leg.getElevationProfile();
         this.totalElevationGained += p.elevationGained();
         this.totalElevationLost += p.elevationLost();
       }
     }
-    this.walkingDuration = totalDuration.minus(transitDuration).minus(nonTransitDuration);
+    this.waitingDuration = totalDuration.minus(transitDuration).minus(nonTransitDuration);
```

Comment:
This reports the street duration, not the waiting duration. The waiting duration is the times between legs